### ISSUE TITLE
[ZF3] [RFC] Removal of `Zend\Loader`

### ISSUE THREAD

**Issue by Ocramius** [zendframework/zendframework#6285]
## Composer as primary autoloader

Just a suggestion, but I think that [composer](http://getcomposer.org/) pretty much took over the autoloading-related problems in all of PHP.

This is a good time to dismiss `Zend\Loader`.

**EDIT**: the `git submodule update --init` approach could also finally die :-)

## Compatibility

Almost every module that is on github currently provides a `composer.json` file. The BC break would be only for most in-house modules that still rely on `Module#getAutoloaderConfig()`.

We can get rid of `getAutoloaderConfig()` or provide a small compat layer that just merges the fetched configuration into the already existing [`Composer\Autoload\ClassLoader`](https://github.com/composer/composer/blob/0c343f925ad578a9da4d9a4cfff4e91b500d206c/src/Composer/Autoload/ClassLoader.php).

## Advantages

One less component to maintain, nuff said.

## Problems

The only issue with composer-first autoloading is security, but I think that can be dealt with until this issue becomes a PR.

Additionally, composer can be used without packagist and as a pure autoloader (`composer.phar dump-autoload`).

**Comment by internalsystemerror** [comment]
In case its relevant, I only ended up learning composer because I wanted to try out ZF2 and I'm forever greatful for it. First thing I ended up doing was abandoning the use of the zf2 autoloader. Since I'm using packages that require composer now I'd rather use 1 autoloader than 2.

One thing that struck my attention was the mention of enterprise systems that move a lot slower. I happen to be working for one of said enterprises where over the past few months I've been introducing the idea of composer and have started to use packages from it. Things do move very slowly indeed although the problems have been reluctance more than anything technical.

In short, those enterprise companies move so slow that by the time they end up wanting to use ZF3, will be when you guys are thinking about ZF4, so maybe isnt as much of a concern as you might think.

Just my two cents, hope it helps.

**Comment by freemany** [comment]
why reinvent the wheel, composer is good<img class="emoji" title=":+1:" alt=":+1:" src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f44d.png" height="20" width="20" align="absmiddle">

**Comment by GeeH** [comment]
This issue has been closed as part of the bug migration program as outlined here - http://framework.zend.com/blog/2016-04-11-issue-closures.html
